force-app/main/default/classes/PMT_RefundService.cls

```diff
@@ -241,10 +234,12 @@ public inherited sharing class PMT_RefundService {
         if (this.refundRecords == null) {
             this.refundRecords = new List<npe01__OppPayment__c>();
         }
+
         this.refundRecords.add(refund);
+        return refund;
     }
 
-    private void updateParentOpportunity(npe01__OppPayment__c originalPayment) {
+    public void updateParentOpportunity(npe01__OppPayment__c originalPayment, Decimal refundAmount) {
```

Comment:
Temporary change this method to a public method. In the next WI, this method will be refactored into the trigger alongside the Allocation update.